flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/CommonCalc.scala

```diff
@@ -171,4 +186,47 @@ trait CommonCalc {
       rowCnt
     }
   }
+
+  /**
+    * split origin generated code to split function calls, only used for calc.
+    * @param generator
+    * @param codeBuffer
+    * @param maxLength
+    * @return (method definitions, method bodies, method callings) of split function calls
+    */
+  private def generateCalcSplitFunctions(
```

Comment:
Seems like in order to avoid the out of compilation limit issue, you need to put this in all Common nodes. 
Any changes we can put it in `plan/util` and make it more generic? will probably imagine same thing could happen to any plan node that requires to generate a row of outputs.